apps/console/src/protected-app.tsx

```diff
@@ -0,0 +1,235 @@
+/**
+ * Copyright (c) 2020, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import {
+    AuthenticatedUserInfo,
+    BasicUserInfo,
+    Hooks,
+    OIDCEndpoints,
+    SPAUtils,
+    SecureApp,
+    useAuthContext
+} from "@asgardeo/auth-react";
+import { AppConstants as CommonAppConstants } from "@wso2is/core/constants";
+import { TenantListInterface } from "@wso2is/core/models";
+import { setSignIn } from "@wso2is/core/store";
+import { AuthenticateUtils as CommonAuthenticateUtils, ContextUtils } from "@wso2is/core/utils";
+import axios from "axios";
+import React, { FunctionComponent, ReactElement, lazy, useEffect } from "react";
+import { useDispatch } from "react-redux";
+import { AuthenticateUtils, getProfileInformation } from "./features/authentication";
+import { Config, HttpUtils, PreLoader } from "./features/core";
+import { AppConstants, CommonConstants } from "./features/core/constants";
+import { history } from "./features/core/helpers";
+
+const AUTHORIZATION_ENDPOINT = "authorization_endpoint";
+const TOKEN_ENDPOINT = "token_endpoint";
+const OIDC_SESSION_IFRAME_ENDPOINT = "oidc_session_iframe_endpoint";
+const LOGOUT_URL = "sign_out_url";
+
+const App = lazy(() => import("./app"));
+export const ProtectedApp: FunctionComponent<Record<string, never>> = (): ReactElement => {
```

Comment:
Would be better to add some doc comments for the component IMO.

Reply:
Resolved by cdf7dba